modules/cardano-faucet-module.nix

```diff
@@ -0,0 +1,268 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.cardano-faucet;
+
+  iohkPkgs = import ../default.nix { inherit config pkgs; inherit (pkgs) system; };
+  walletPackage = iohkPkgs.cardano-sl-wallet-new-static;
+  faucetPackage = iohkPkgs.cardano-sl-faucet-static;
+  toolsPackage = iohkPkgs.cardano-sl-tools;
+  configFiles = iohkPkgs.cardano-sl-config;
+
+  walletStateDir = "${cfg.home}/wallet";
+  tlsDir = "${cfg.home}/tls";
+  walletPort = 8090;
+  walletUseTLS = true;
+  faucetEkgPort = 8001;  # currently hardcoded in faucet
+  walletEkgPort = 8002;
+
+  defaultLoggingCfg = pkgs.writeText "logging.cfg" ''
+    termSeveritiesErr: All
+    loggerTree:
+      severity: Info+      # severities for root logger
+      withdraw:            # logger for the withdraw action
+        severity: Debug+   # severities withdraw logger
+  '';
+
+  cfgFile = pkgs.writeText "config.json" (builtins.toJSON ({
+    wallet-host = "127.0.0.1";
+    wallet-port = walletPort;
+    inherit (cfg) port;
+    payment-distribution = {
+      mean = cfg.paymentAmountMean;
+      scale = cfg.paymentAmountScale;
+    };
+    logging-config = cfg.loggingConfigFile;
+    source-wallet.generate-to = "${cfg.home}/generated-wallet.json";
+  } // (optionalAttrs (cfg.recaptcha != null) {
+    # fixme: recaptcha secret in file
+    recaptcha-secret = cfg.recaptcha.secretKey;
```

Comment:
good point on public vs secret recaptcha keys, but this is where the secret is getting dumped to the nix store.

Reply:
Faucet has now been fixed to read the secret key from file so I can change this.